### PR TITLE
API: Block | Get list of block

### PR DESCRIPTION
**Pull request by Sotatek-GiangNguyen7**

### CODE REVIEW

File: src/test/java/tests/block/BlockListTests.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+        String blockId = "950073";
+        blockListModel = (BlockListModel) blockSteps
+                .getTxListOfBlock(blockId)
+                .validateStatusCode(HttpURLConnection.HTTP_OK)
+                .saveResponseObject(BlockListModel.class);
+
+        blockSteps.then_verifyDataNull(blockListModel);
+    }
+    @Test(description = "get block list unsuccessfully with invalid page", groups = {"block","block-list"}, dataProvider = "paramInvalidPage")
```
Comment (ghost) [inline src/test/java/tests/block/BlockListTests.java]: should be "get block list successfully with invalid page"

File: src/test/java/tests/block/BlockListTests.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    public void getBlockListUnsuccessfullyWithInvalidPage(Object page){
```
Comment (ghost) [inline src/test/java/tests/block/BlockListTests.java]: and here

File: src/test/java/tests/block/BlockListTests.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    @DataProvider(name ="paramInvalidPage")
+    public Object[][] dataSetInvalidPage(){
+        return new Object[][]{
+                {"abc"},
+                {"-10"},
+                {" "},
+                {"@#$%%"},
+        };
+    }
+    @Test(description = "get block list unsuccessfully with invalid blockId", groups = {"block","block-list"}, dataProvider = "paramInvalidBlockId")
```
Comment (ghost) [inline src/test/java/tests/block/BlockListTests.java:61]: and here

File: src/test/java/tests/block/BlockListTests.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    public void getBlockListUnsuccessfullyWithInvalidBlockId(Object blockId){
```
Comment (ghost) [inline src/test/java/tests/block/BlockListTests.java:62]: and here

File: src/test/java/tests/block/BlockListTests.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    @DataProvider(name ="paramInvalidBlockId")
+    public Object[][] dataSetInvalidBlockId(){
+        return new Object[][]{
+                {"@#$"},
+                {" "},
+                {"abcd"},
+                {"12345"},
+        };
+    }
+    @Test(description = "get block list unsuccessfully with invalid size", groups = {"block","block-list"}, dataProvider = "paramInvalidSize")
```
Comment (ghost) [inline src/test/java/tests/block/BlockListTests.java]: and here

File: src/test/java/tests/block/BlockListTests.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    public void getBlockListUnsuccessfullyWithInvalidSize(Object size){
```
Comment (ghost) [inline src/test/java/tests/block/BlockListTests.java]: and here

File: src/test/java/tests/block/BlockListTests.java (hunk 3 above)
Comment (Sotatek-GiangNguyen7) [reply src/test/java/tests/block/BlockListTests.java:61]: this function is going to give null data so i think it should name unsuccess

File: src/test/java/tests/block/BlockListTests.java (hunk 4 above)
Comment (Sotatek-GiangNguyen7) [reply src/test/java/tests/block/BlockListTests.java:62]: this function is going to give null data so i think it should name unsuccess

**Comment by Sotatek-GiangNguyen7** [comment]
image

**Review by ghost — APPROVED** [review]

**Pull request merged**